connect/runtime/src/main/java/org/apache/kafka/connect/runtime/AbstractHerder.java

```diff
@@ -486,21 +488,43 @@ ConfigInfos validateConnectorConfig(Map<String, String> connectorProps, boolean
             Set<String> allGroups = new LinkedHashSet<>(enrichedConfigDef.groups());
 
             // do custom connector-specific validation
-            ConfigDef configDef = connector.config();
+            ConfigDef configDef;
+            try {
+                configDef = connector.config();
+            } catch (Throwable e) {
+                throw new ConnectException(
+                        String.format(
+                                "unable to evaluate %s.config()",
+                                connector.pluginClass().getName()
+                        ),
+                        e
+                );
+            }
```

Comment:
Would it be cleaner to add a `throws Exception` clause to this variant of `validateConnectorConfig`? Right now it's only called from [one (non-testing) location](https://github.com/apache/kafka/blob/7322f4cd55dc08abdc6ccf51ed33f7f0d869dd0e/connect/runtime/src/main/java/org/apache/kafka/connect/runtime/AbstractHerder.java#L382-L387), which already has the necessary try/catch wrapper.

The message also doesn't seem terribly useful here, considering the method and class name are already going to be present in the stack trace for the exception.